compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1818,9 +1819,17 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
    */
   def tryInsertApplyOrImplicit(tree: Tree, pt: ProtoType)(fallBack: => Tree)(implicit ctx: Context): Tree = {
 
+    def isSyntheticApply(tree: Tree): Boolean = tree match {
+      case tree: Select => tree.getAttachment(InsertedApply).isDefined
```

Comment:
One reason to keep the current PR is that the first `apply` could be macro generated where the macro returns a map, say. Then it would seem sensical to insert the 2nd apply.